docs/app/_components/SiteMenu.tsx

```diff
@@ -66,33 +181,119 @@ export const SiteMenu = () => {
+            if (searchValue.toLowerCase().includes(query.toLowerCase()))
+              return 1;
+            return 0;
+          }}
+          onKeyDown={e => {
+            if (e.key === 'Escape' || (e.key === 'Backspace' && !query)) {
+              e.preventDefault();
+              setPages([]);
+            }
+          }}
+        >
           <div className="flex items-center gap-1.5 border-b px-3">
-            <Search className="size-4 opacity-50"></Search>
-            <Command.Input
+            <Search className="size-4 opacity-50" />
+            <CustomInput
               value={query}
-              autoFocus
               onValueChange={setQuery}
-              placeholder="Type to search ..."
-              className="placeholder:text-text-primary-muted h-11 w-full bg-transparent outline-none"
+              onHandlePages={handlePages}
             />
           </div>
           <Command.List className="scrollbar-thin scrollbar-thumb-slate-400 scrollbar-track-transparent scrollbar-thumb-rounded-full max-h-[300px] overflow-y-auto overflow-x-hidden">
             <Command.Empty className="py-6 text-center text-sm">
               No results found.
             </Command.Empty>
             {groupedPages.map(({ name, items }) => (
+              <CommandGroup
+                heading={name}
+                key={name}
+                className={classNames.section}
+              >
+                {items.map(page => (
+                  <>
+                    <Command.Item
+                      className={classNames.item}
+                      key={page.slug}
+                      value={page.slug}
+                      onSelect={() => goto(page.slug)}
+                    >
+                      <Inline space={4} alignY="center">
+                        {page.title}
+                        <Split />
+                        <Hotkey letter="D" />
+                      </Inline>
+                    </Command.Item>
+                    {subPage === page.slug && (
+                      <>
+                        {Object.values(page.headings).map(
+                          (sub: { slug: string; text: string }) => (
+                            <Command.Item
+                              value={`${page.slug}${sub.slug}`}
+                              className={cn(
+                                'text-text-primary-muted ml-7',
+                                classNames.item
+                              )}
+                              onSelect={() => goto(`${page.slug}#${sub.slug}`)}
+                            >
+                              {sub.text}
+                            </Command.Item>
+                          )
+                        )}
+                      </>
+                    )}
+                  </>
+                ))}
+              </CommandGroup>
+            ))}
+            {/* update themes command */}
+            {themeswitch.map(({ name, items }) => (
```

Comment:
Maybe pull these out into their own components? this would maek this part more readable.

Reply:
You mean every different item should have its own component?

Reply:
Yes something like that e.g. `<ThemeComppands commands={themeswitch}/>`

Reply:
I moved it into a new file, can you have a look again?